modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java

```diff
@@ -777,10 +777,20 @@ public Boolean isRequireAuthentication() throws PortalException {
 	}
 
 	public boolean isSharedURL() {
+		DDMFormInstance ddmFormInstance = getFormInstance();
 		ThemeDisplay themeDisplay = getThemeDisplay();
 
-		return StringUtil.contains(
-			themeDisplay.getURLCurrent(), "shared", StringPool.BLANK);
+		if (StringUtil.contains(
+				themeDisplay.getURLCurrent(), "shared", StringPool.BLANK) &&
+			StringUtil.contains(
+				themeDisplay.getURLCurrent(),
+				String.valueOf(ddmFormInstance.getFormInstanceId()),
+				StringPool.BLANK)) {
+
+			return true;
+		}
+
+		return false;
```

Comment:
@Marina-Gouveia please check if this change will impact the existing [test](https://github.com/liferay/liferay-portal/blob/16e66444b76c8bb1a39aba9a65327dba06746edd/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContextTest.java#L668-L687)